packages/features/ee/dsync/lib/handleUserEvents.ts

```diff
@@ -25,7 +26,7 @@ const handleUserEvents = async (event: DirectorySyncEvent, organizationId: numbe
   });
 
   // User is already a part of that org
-  if (user?.organizationId && eventData.active) {
+  if (user && UserRepository.isAMemberOfOrganization({ user, organizationId }) && eventData.active) {
```

Comment:
There was a lot of errors around creating profiles for existing org members so this should fix that logic